### PR TITLE
[FLINK] add option for flink job listener to read from flink conf

### PR DESCRIPTION
**Pull request by ensctom**
### Problem

Currently there is no way to pass in jobname and namespace except explicitly defining them when using the job builder.

Closes: [#2230](https://github.com/OpenLineage/OpenLineage/issues/2230)

### Solution

Since we are already passing in the flink execution environment to the builder. It would be nice to set the values of namespace and job name via reading the execution environment configuration if not already set by the builder.

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [x] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

File: integration/flink/src/main/java/io/openlineage/flink/OpenLineageFlinkJobListener.java [hunk 1]
```diff
@@ -44,10 +46,28 @@ public class OpenLineageFlinkJobListener implements JobListener {
 
   public static final String DEFAULT_JOB_NAMESPACE = "flink_jobs";
 
-  /** Default duration between REST API calls for checkpoint stats. */
   static final Duration DEFAULT_TRACKING_INTERVAL = Duration.ofSeconds(10);
 
+  public static final ConfigOption<String> OPEN_LINEAGE_LISTENER_CONFIG_JOB_NAMESPACE =
```
Comment (pawel-big-lebowski) [inline integration/flink/src/main/java/io/openlineage/flink/OpenLineageFlinkJobListener.java:51]: Would it be doable and implement that in a similar manner as here: https://github.com/OpenLineage/OpenLineage/blob/1.5.0/integration/spark/app/src/main/java/io/openlineage/spark/agent/ArgumentParser.java#L82C7-L82C7 ?

For Apache Spark, we are translating `spark.conf` entries to `OpenLineageYaml`. The great advantage of this approach is that, whenever a new parameter is added to `openlineage-java`, it is automatically available within Spark integration by setting a proper config. 

My concern is that the list of config params will grow.

Comment (ensctom) [reply integration/flink/src/main/java/io/openlineage/flink/OpenLineageFlinkJobListener.java:51]: Thanks for the suggestion.

Then would it make sense to move the job name, namespace parameters inside `OpenLineageYaml` directly?

Then in the builder we can use `FlinkConfigParser.parse(configuration)`  to create an instance of OpenLineageYaml inside the OpenlineageFlinkJobListener?


Please let me know if I misunderstood.

Comment (pawel-big-lebowski) [reply integration/flink/src/main/java/io/openlineage/flink/OpenLineageFlinkJobListener.java:51]: My thinking was wrong. Transport config can be already read from flink conf which was introduced with this PR: https://github.com/OpenLineage/OpenLineage/pull/2033

Additionally, `jobName` and `jobNamespace` has to be set directly in flink integration when OL event is built. 

So, my suggestion is wrong. Sorry for that.

**Review by pawel-big-lebowski — APPROVED** [review]
👍 💯

**Comment by pawel-big-lebowski** [comment]
@ensctom could u update changelog and add some info in doc: https://github.com/OpenLineage/OpenLineage/tree/main/integration/flink ?

**Comment by pawel-big-lebowski** [comment]
Hi @ensctom, we did have an issue with CI, which got fixed in https://github.com/OpenLineage/OpenLineage/pull/2235. 
Would you mind rebasing your PR onto current main? Sry for inconvenience.

**Comment by ensctom** [comment]
> Hi @ensctom, we did have an issue with CI, which got fixed in #2235. Would you mind rebasing your PR onto current main? Sry for inconvenience.

I merged from main and the CI passes now. I had to merge from main instead of rebasing because on windows machine one of the files kept auto changing which prevented me from rebasing.

**Pull request merged**